src/components/tx-flow/index.tsx

```diff
@@ -54,14 +57,13 @@ export const TxModalProvider = ({ children }: { children: ReactNode }): ReactEle
 
   // Show the modal if user navigates
   useEffect(() => {
-    if (!txFlow) {
+    const shouldWarn = txFlow && txFlow.type !== newTxMenu.type
```

Comment:
```suggestion
    const shouldWarn = txFlow && !(txFlow.type instanceof NewTxMenu)
```

This also works I believe and we don't line 21 then.